lib/perl/Genome/Model/Command/Admin/RemoveDiskAllocationsFromTestdb.pm

```diff
@@ -0,0 +1,308 @@
+use Genome::Config;
+use Try::Tiny;
+use Sub::Install qw();
+use Sub::Name qw();
+
+use constant KB_IN_ONE_GB => 1024 * 1024;
+use constant KB_IN_ONE_MB => 1024;
+
+class Genome::Model::Command::Admin::RemoveDiskAllocationsFromTestdb {
+    is => 'Command::V2',
+    doc => 'Remove disk allocations that were created while running under a test database',
+    has => [
+        database_name => {
+            is => 'Text',
+            default_value => _get_default_database_name(),
+            doc => 'test database name, derived from the ds_gmschema_server config value',
+        },
+        database_server => {
+            is => 'Text',
+            default_value => _get_default_database_server(),
+            doc => 'database server name, derived from the ds_gmschema_server config value',
+        
+        },
+        database_port => {
+            is => 'Text',
+            default_value => _get_default_database_port(),
+            doc => 'database listening port, derived from the ds_gmschema_server config value',
+        },
+        template_name => {
+            is => 'Text',
+            default_value => _get_default_template_name(),
+            doc => 'template database name, resolved from the database host in the ds_gmschema_server config value',
+        },
+    ],
+};
+
+sub _get_default_database_name {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return $conn->{dbname};
+}
+
+sub _get_default_database_server {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return $conn->{host};
+}
+
+sub _get_default_database_port {
+    my $conn = _parse_database_connection_info_from_env_var();
+    return( $conn->{port} // 5432 );
+}
+
+sub _get_default_template_name {
+    my $test_db_name = __PACKAGE__->_get_default_database_name();
+    return __PACKAGE__->get_template_name_for_database_name($test_db_name);
+}
+
+sub _parse_database_connection_info_from_env_var {
+    my %connection;
+    foreach my $key ( qw( dbname host port ) ) {
+        no warnings 'uninitialized';
+        my $server = Genome::Config::get('ds_gmschema_server');
+        ($connection{$key}) = $server =~ m/$key=(.*?)(?:;|$)/;
+    }
+    return \%connection;
+}
+
+sub execute {
+    my $self = shift;
+
+    unless ($self->is_running_in_test_env) {
+        die $self->error_message('Must be run within a test environment created with genome-test-env');
+    }
+
+    my @allocations = $self->collect_newly_created_allocations();
+    $self->report_allocations_to_delete(@allocations);
+    $self->delete_allocations(@allocations);
+    return 1;
+}
+
+sub is_running_in_test_env {
+    my $self = shift;
+
+    return Genome::Config::get('testing');
+}
+
+sub collect_newly_created_allocations {
+    my $self = shift;
+
+    my $tmpl_allocations = $self->_make_iterator_for_template_allocations();
+    my $db_allocations = $self->_make_iterator_for_database_allocations();
+
```

Comment:
Holy cats we are going to have to be careful with this.  By comparing the test db to the template rather than 'production', one danger would be that someone (e.g. me) would have gone into the test db and possible mucked around, maybe using fill_db to load some extra things from production, etc. etc. This would delete those things.  Maybe the answer to this is "Allison don't muck around with test_filldb!"  Just something to keep in mind.